piccolo3/server/PiccoloCoolboxControl.py

```diff
@@ -67,7 +68,7 @@ def initialize_serial(self):
             self.log.debug("Serial error:", e)
 
     def initialise_coolbox(self):
-        with self._serial_lock:
+        with (yield from self._serial_lock):
```

Comment:
I think this should be
```
await with self._serial_kock:
```

Reply:
That wasn't it, but it might have led me to the correct pattern. I can get past init with 
```
async def initialise_coolbox(self):
        async with self._serial_lock:
```
Although without a coolbox and piccolo it's hard to say whether this works properly or just doesn't throw an exception!

Reply:
since the initialise_coolbox method is a coroutine you will need to schedule a task for  it, ie in __init__
```
loop = asyncio.get_event_loop()
task = loop.create_task(self.initialise_coolbox())
```
not sure if this will work, but something like that might

Reply:
I think [As far as I can see without a coolbox to test against] this worked, thanks!